modules/configuration/src/test/java/org/apache/ignite/internal/configuration/RenamedConfigurationTest.java

```diff
@@ -0,0 +1,279 @@
+    }
+
+    @Test
+    public void testLegacyNameIsRecognisedOnStartup() {
+        // Default value was set when registry was started with old configuration.
+        assertThat(
+                registry.getConfiguration(RenamedTestNewConfiguration.KEY).newInnerName().newName().value(),
+                Matchers.equalTo(OLD_DEFAULT)
+        );
+
+        assertThat(storage.readLatest(String.format("key.%s.%s", OLD_INNER_NAME, OLD_VALUE_NAME)), willBe(nullValue()));
+    }
+
+    @Test
+    public void testLegacyNameIsRecognisedOnUpdate() {
+        String updatedValue = "updatedValue";
+        String configWithFirstLegacyName = String.format("key.%s.%s = \"%s\"", OLD_INNER_NAME, OLD_VALUE_NAME, updatedValue);
+        updateConfig(registry, configWithFirstLegacyName);
+
+        assertThat(
+                registry.getConfiguration(RenamedTestNewConfiguration.KEY).newInnerName().newName().value(),
+                Matchers.equalTo(updatedValue)
+        );
+        assertThat(storage.readLatest(String.format("key.%s.%s", OLD_INNER_NAME, OLD_VALUE_NAME)), willBe(nullValue()));
+
+        String secondUpdatedValue = "secondUpdatedValue";
+        String configWithSecondLegacyName = String.format("key.%s.%s = \"%s\"", SECOND_OLD_INNER_NAME, OLD_VALUE_NAME, secondUpdatedValue);
+
+        updateConfig(registry, configWithSecondLegacyName);
+
+        assertThat(
+                registry.getConfiguration(RenamedTestNewConfiguration.KEY).newInnerName().newName().value(),
+                Matchers.equalTo(secondUpdatedValue)
+        );
+        assertThat(storage.readLatest(String.format("key.%s.%s", SECOND_OLD_INNER_NAME, OLD_VALUE_NAME)), willBe(nullValue()));
+    }
+
+    @Test
+    public void testNewValuePersistsAfterRestart() {
+        String updatedValue = "updatedValue";
+        String updatedConfig = String.format("key.%s.%s = \"%s\"", OLD_INNER_NAME, OLD_VALUE_NAME, updatedValue);
+        updateConfig(registry, updatedConfig);
+
+        assertThat(
+                registry.getConfiguration(RenamedTestNewConfiguration.KEY).newInnerName().newName().value(),
+                Matchers.equalTo(updatedValue)
+        );
+
+        stopRegistry(registry);
+        registry = startRegistry(RenamedTestNewConfiguration.KEY, NEW_GENERATOR);
+
+        assertThat(storage.readLatest(String.format("key.%s.%s", OLD_INNER_NAME, OLD_VALUE_NAME)), willBe(nullValue()));
+
+        assertThat(
+                registry.getConfiguration(RenamedTestNewConfiguration.KEY).newInnerName().newName().value(),
+                Matchers.equalTo(updatedValue)
+        );
+    }
+
+    @Test
+    public void testNamedListLegacyNameIsRecognisedOnUpdate() {
+        registry = startRegistry(RenamedTestNewConfiguration.KEY, NEW_GENERATOR);
+
+        String instanceName = "listInstance";
+        String newValue = "newValue";
+
+        String updatedConfig = String.format("key.%s.%s.%s=\"%s\"", OLD_LIST_NAME, instanceName, OLD_VALUE_NAME, newValue);
```

Comment:
I think that you don't really have to quote the `%s` at the end, let's avoid all unnecessary noise